### PR TITLE
Sundials no longer needs to ship its own suitesparse libraries

### PR DESCRIPTION
**Pull request by ViralBShah**
The Julia SuiteSparse build now includes KLU, which is what Sundials wants.

### CODE REVIEW

**Comment by ViralBShah** [comment]
Do I need to have a suitesparse dependency here explicitly, or will it just pick it up from julia? I see we specify `openblas_jll`, but IIRC, something special was done to allow that.

**Comment by ViralBShah** [comment]
The issue is with building `libsundials_sunlinsolklu`.

**Comment by ViralBShah** [comment]
@giordano Would love it if you could suggest what I need to do.

**Comment by giordano** [comment]
Yes, you need to specify the new dependency, but we need first to update the builder.  I'll do that in a few minutes

**Comment by jd-lara** [comment]
@ViralBShah my understanding is that Sundials needs not only KLU but needs other libraries from SuiteSparse https://github.com/Energy-MAC/SundialsBuilder/blob/9b04e902454f225942c8021a944acf31ef1384f7/build_tarballs.jl#L42

**Comment by ViralBShah** [comment]
@jd-lara We are building all of those.

File: S/Sundials/build_tarballs.jl [hunk 1]
```diff
@@ -62,20 +42,12 @@ fi
...
-    cp -L $WORKSPACE/destdir/lib/*.dll $WORKSPACE/destdir/bin
-    rm -f $WORKSPACE/destdir/lib/*.dll
-fi
 """
 
 # We attempt to build for all defined platforms
 platforms = supported_platforms()
 
 products = [
-    LibraryProduct("libbtf", :libbtf),
```
[the remark this answers is not in the document]
Comment (ViralBShah) [reply S/Sundials/build_tarballs.jl]: That's right.

**Comment by ViralBShah** [comment]
@giordano It seems that I can't get it to find the BLAS and LAPACK. Any ideas what's going on?

Also, can you tell what might be causing the w64 builds to fail?

File: S/Sundials/build_tarballs.jl [hunk 2]
```diff
@@ -7,80 +7,46 @@ version = v"3.1.1"
...
-CMAKE_FLAGS="${CMAKE_FLAGS} -DENABLE_LAPACK=ON -DLAPACK_LIBRARIES=$LIBBLAS"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DBLAS_ENABLE=ON -DLAPACK_ENABLE=ON"
 
-if [[ ${nbits} == 32 ]]; then
-    echo "***   32-bit BUILD   ***"
-    LIBBLAS="$prefix/lib/libopenblas.so"
-    cmake ${CMAKE_FLAGS} -DBLAS_LIBRARIES=\"$LIBBLAS\" -DSUNDIALS_INDEX_TYPE=int32_t ..
+if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
+    patch -p0 < $WORKSPACE/srcdir/patches/Sundials_ilp64.patch
+    CMAKE_FLAGS="${CMAKE_FLAGS} -DBLAS_LIBRARIES=\"-L$prefix/lib -lopenblas64_\" -DLAPACK_LIBRARIES=\"-L$prefix/lib -lopenblas64_\""
```
Comment (staticfloat) [inline S/Sundials/build_tarballs.jl]: The issue seems to be quote-escaping; I'm going to try changing `\"` to `\\\"` instead.

File: S/Sundials/build_tarballs.jl [hunk 3]
```diff
@@ -7,80 +7,47 @@ version = v"3.1.1"
...
+CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release -DEXAMPLES_ENABLE_C=OFF)
+CMAKE_FLAGS+=(-DKLU_ENABLE=ON -DKLU_INCLUDE_DIR="$prefix/include" -DKLU_LIBRARY_DIR="$libdir")
+CMAKE_FLAGS+=(-DBLAS_ENABLE=ON -DLAPACK_ENABLE=ON)
+
+if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
     patch -p0 < $WORKSPACE/srcdir/patches/Sundials_ilp64.patch
+    CMAKE_FLAGS+=(-DBLAS_LIBRARIES="-L${libdir} -lopenblas64_" -DLAPACK_LIBRARIES="-L${libdir} -lopenblas64_")
+else
+    CMAKE_FLAGS+=(-DBLAS_LIBRARIES="-L${libdir} -lopenblas" -DLAPACK_LIBRARIES="-L${libdir} -lopenblas")
+    BLAS="-L${libdir} -lopenblas"
```
Comment (giordano) [inline S/Sundials/build_tarballs.jl]: I think this variable isn't used anywhere, is it?

Comment (ViralBShah) [reply S/Sundials/build_tarballs.jl]: That's right. It can be removed.

**Pull request merged**